components/classroom/fetchData.mjs

```diff
@@ -1,32 +1,33 @@
-    const projectId = document.getElementById('projectId').value;
-    const errorMsg = document.getElementById('error-msg');
+const PROJECT_FORM = document.getElementById('projectForm');
 
-    try {
-        const response = await fetch(`https://dev.api.t-pen.org/project/${projectId}`, {
-            method: 'GET',
-            headers: {
-                'Authorization': `Bearer ${token}`
-            }
-        });
+PROJECT_FORM.TPEN = new TPEN();
+TPEN.attachAuthentication(PROJECT_FORM);
 
-        if (!response.ok) {
-            throw new Error('Project not found or access denied');
+if (PROJECT_FORM.TPEN.activeProject?._id) {
+    projectId.value = PROJECT_FORM.TPEN.activeProject._id;
+    const token = TPEN.getAuthorization();
+    // until the Project class is better defined, we'll just fetch the project data directly
+    const project = await fetch(`${PROJECT_FORM.TPEN.servicesURL}/project/${PROJECT_FORM.TPEN.activeProject._id}`, {
+        method: 'GET',
+        headers: {
+            'Authorization': `Bearer ${token}`
         }
-        
-        const projectData = await response.json();
-        displayProjectData(projectData);
-        window.location.href = 'rabc.html';
-    } catch (error) {
-        errorMsg.style.display = 'block';
-    }
+    })
+        .then(res => res.ok ? res.json() : Promise.reject(res.status))
+        .then(data => msg.innerHTML = `<pre>${JSON.stringify(data, null, 2)}</pre>`)
+        .catch(err => msg.innerHTML = `<pre>${JSON.stringify(err, null, 2)}</pre>`);
+}
+
+PROJECT_FORM.addEventListener('submit', async function (event) {
+    const projectID = projectId.value;
+    location.href = '?projectID=' + projectID;  // Redirect to the same page with the projectID in the query string
```

Comment:
This just loads the page with the queryString, so the processing happens naturally.